test/L1ERC20Bridge.t.sol

```diff
@@ -3,33 +3,159 @@ pragma solidity ^0.8.0;
+contract Constructor is Test, Constants {
+  function testFork_CorrectlySetAllArgs(address l1Erc) public {
+    FakeERC20 fake = new FakeERC20("Hello", "WRLD");
+    IGovernor gov = new GovernorMock("Testington Dao", fake);
+    L1ERC20Bridge bridge =
+    new L1ERC20Bridge(address(l1Erc), wormholeCoreFuji, address(gov), wormholeFujiId, wormholePolygonId);
+    assertEq(address(bridge.L1_TOKEN()), l1Erc, "L1 token is not set correctly");
+  }
+}
+
+contract Initialize is L1ERC20BridgeTest {
+  function testFork_CorrectlyInitializeL2Token(address l2Erc20) public {
+    bridge.initialize(address(l2Erc20));
+    assertEq(bridge.L2_TOKEN_ADDRESS(), l2Erc20, "L2 token address is not setup correctly");
+    assertEq(bridge.INITIALIZED(), true, "Bridge isn't initialized");
+  }
+
+  function testFork_InitializeL2AddressWhenAlreadyInitialized(address l2Erc20) public {
+    bridge.initialize(address(l2Erc20));
+
+    vm.expectRevert(L1ERC20Bridge.AlreadyInitialized.selector);
+    bridge.initialize(address(l2Erc20));
   }
 }
 
 contract Deposit is L1ERC20BridgeTest {
-  function testFork_CorrectlyDepositTokens() public {
-    FakeERC20 erc20 = new FakeERC20("Hello", "WRLD");
-    IGovernor gov = new GovernorMock("Testington Dao", erc20);
-    L1ERC20Bridge bridge = new L1ERC20Bridge(address(erc20), wormholeCoreFuji, address(gov));
-    bridge.initialize(0xBaA85b5C4c74f53c46872acfF2750f512bcBEC43);
-
-    erc20.approve(address(bridge), 100_000);
-    erc20.mint(address(this), 100_000);
+  function testFork_CorrectlyDepositTokens(uint96 _amount) public {
+    bridge.initialize(address(l2Erc20));
+    uint256 cost = bridge.quoteDeliveryCost(wormholePolygonId);
+    vm.recordLogs();
+
+    fake.approve(address(bridge), _amount);
+    fake.mint(address(this), _amount);
+    vm.deal(address(this), 1 ether);
+
+    bridge.deposit{value: cost}(address(this), _amount);
+
+    vm.prank(wormholeCoreMumbai);
+    performDelivery();
+
+    vm.selectFork(targetFork);
+    assertEq(l2Erc20.balanceOf(address(this)), _amount, "L2 token balance is not correct");
+  }
+}
+
+// Top level receive is tested in L2ERC20 and L2VoteAggregator
+contract _ReceiveWithdrawalWormholeMessages is Test, Constants {
+  function testForkFuzz_CorrectlyReceiveWithdrawal(
+    address _account,
+    uint96 _amount,
+    address l2Erc20
+  ) public {
+    vm.assume(_account != address(0));
+    FakeERC20 fake = new FakeERC20("Hello", "WRLD");
+    IGovernor gov = new GovernorMock("Testington Dao", fake);
+    L1ERC20BridgeHarness bridge =
+    new L1ERC20BridgeHarness(address(fake), wormholeCoreFuji, address(gov), wormholeFujiId, wormholePolygonId);
+
+    bridge.initialize(address(l2Erc20));
+    fake.approve(address(this), _amount);
+    fake.mint(address(this), _amount);
     vm.deal(address(this), 1 ether);
 
-    bridge.deposit(address(this), 100_000);
+    fake.transfer(address(bridge), _amount);
+    assertEq(fake.balanceOf(address(bridge)), _amount, "The Bridge balance is incorrect");
```

Comment:
I don't think this assertion is testing anything useful.

Reply:
I sort of agree. It can be removed, but having it verifies that the setup is in an expected state. We have similar checks in Pooltogether and Gitcoin. I definitely can remove, but I also don't see the harm in leaving it.

https://github.com/ScopeLift/pooltogether-governor-upgrade/blob/main/test/helpers/ProposalTest.sol#L264

Reply:
Fair enough, can leave as-is